### PR TITLE
[cilium] setting VXLAN port into allowed range

### PR DESCRIPTION
**Pull request by Beastlex**
## Description
Setting VXLAN port into allowed range

## Why do we need it, and what problem does it solve?
This will make it easier for administrators to create network policies on firewalls within the corporate infrastructure. They will be able to add a single range to the policy.

## What is the expected result?
* Existing cluster installations will remain on ports 8469 or 8472 (with virtualization). 
* In new installations, the ports will be set to 4299 or 4298 (with virtualization).

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cilium
type: chore
summary: Set VXLAN port into allowed range.
impact: In new installations, the Cilium VXLAN ports will be set to 4299 or 4298 (with virtualization).
impact_level: default
```

### CODE REVIEW

File: modules/021-cni-cilium/hooks/set_vxlan_allowed_ports.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"d8-cni-cilium"},
+				},
+			},
+			ExecuteHookOnEvents:          pointer.Bool(false),
+			ExecuteHookOnSynchronization: pointer.Bool(false),
+			FilterFunc:                   filterConfigMap,
+		},
+		{
+			Name:       "agent-daemonset",
```
Comment (apolovov) [inline modules/021-cni-cilium/hooks/set_vxlan_allowed_ports.go:93]: Why don't you use the cilium-configmap to discover if cilium was deployed before?

File: modules/021-cni-cilium/hooks/set_vxlan_allowed_ports_test.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2023 Flant JSC
```
Comment (apolovov) [inline modules/021-cni-cilium/hooks/set_vxlan_allowed_ports_test.go:2]:
```suggestion
Copyright 2024 Flant JSC
```

**Review by apolovov — CHANGES_REQUESTED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/eks'...

File: modules/021-cni-cilium/monitoring/prometheus-rules/config.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      severity_level: "4"
+      tier: application
+    annotations:
+      plk_markup_format: "markdown"
+      plk_protocol_version: "1"
+      plk_create_group_if_not_exists__main: "ClusterHasCniCiliumNonStandardVXLANPort,tier=~tier,prometheus=deckhouse,kubernetes=~kubernetes"
+      plk_grouped_by__main: "ClusterHasCniCiliumNonStandardVXLANPort,tier=~tier,prometheus=deckhouse,kubernetes=~kubernetes"
+      summary: There is non-standard VXLAN port in Cilium config
+      description: |
+        There is non-standard VXLAN port in Cilium config: `{{$labels.port}}` does not fall within the recommended range (4298, 4299).
```
Comment (apolovov) [inline modules/021-cni-cilium/monitoring/prometheus-rules/config.yaml:16]:
```suggestion
        There is non-standard VXLAN port in Cilium config: `{{$labels.port}}` does not fit the recommended range (4298 if the virtualization module enabled or 4299 for regular deckhouse setup).
```

File: modules/021-cni-cilium/monitoring/prometheus-rules/config.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+        It is recommended to check `tunnel-port` parameter in ConfigMap `cilium-configmap` in `d8-cni-cilium` namespace.
```
Comment (apolovov) [inline modules/021-cni-cilium/monitoring/prometheus-rules/config.yaml:18]:
```suggestion
        Consider configuring the `tunnel-port` parameter in `cilium-configmap` ConfigMap (`d8-cni-cilium` namespace) according the recommended range. If you know why you need the non-standard port, just ignore the alert.
```

File: modules/021-cni-cilium/hooks/set_vxlan_allowed_ports.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+		On: { // Virtualization is on
+			TransitionRule{source: 0, target: 4298},
+			TransitionRule{source: 4298, target: 4298},
+		},
+		Off: { // Virtualization is off
+			TransitionRule{source: 0, target: 4299},
+			TransitionRule{source: 4299, target: 4299},
+		},
+	},
+}
```
Comment (apolovov) [inline modules/021-cni-cilium/hooks/set_vxlan_allowed_ports.go:73]:
```suggestion
	Existing: { // existing installation (ConfigMap exists)
		On: { // Virtualization is on
		 	// cm has configured 8469 port, will leave it as is
			TransitionRule{source: 8469, target: 8469},
			
		 	// dreamy case — virtualization was enabled with upgrading d8 simultaneously
			TransitionRule{source: 0, target: 8469},
			
			// dreamy case — virtualization was enabled with upgrading d8 simultaneously and
			// someone configured the port for setup without virtualization 8472 manually, will set the right one
			TransitionRule{source: 8472, target: 4298},
			
			// virtualization module was enabled on regular setup with the right port, will set the 4298
			TransitionRule{source: 4299, target: 4298},
			
			// regular setup with enabled virtualization module and right port, will leave it as is
			TransitionRule{source: 4298, target: 4298},
			
			// if the "source" port is non-standard and didn't mention here, will leave it as is and fire the alert
		},
		Off: { // Virtualization is off
			// our previous standard setup, will set the old default port explicitly
			TransitionRule{source: 0, target: 8472},
			
			// our previous standard setup with explicitly configured 8472 port, will leave the 8472
			TransitionRule{source: 8472, target: 8472},
			
			// regular setup with standard 4299 port, will leave it as is
			TransitionRule{source: 4299, target: 4299},
		},
	},
	New: { // new installation (ConfigMap does not exist)
		On: { // Virtualization is on
			TransitionRule{source: 0, target: 4298},
		},
		Off: { // Virtualization is off
			TransitionRule{source: 0, target: 4299},
		},
	},
}
```

**Review by apolovov — CHANGES_REQUESTED** [review]

**Review by apolovov — CHANGES_REQUESTED** [review]

**Review by apolovov — APPROVED** [review]

File: modules/021-cni-cilium/hooks/set_vxlan_allowed_ports.go [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+		},
+		Off: { // Virtualization is off
+			TransitionRule{source: 0, target: 4299},
+		},
+	},
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 10},
+	Queue:        "/modules/cni-cilium/set-vxlan-allowed-ports",
```
Comment (apolovov) [inline modules/021-cni-cilium/hooks/set_vxlan_allowed_ports.go]:
```suggestion
	Queue:        "/modules/cni-cilium/discover_vxlan_port",
```

File: modules/021-cni-cilium/hooks/discover_vxlan_port_test.go [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+package hooks
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/deckhouse/deckhouse/testing/hooks"
+)
+
+var _ = Describe("Modules :: cni-cilium :: hooks :: set_vxlan_allowed_ports ::", func() {
```
Comment (apolovov) [inline modules/021-cni-cilium/hooks/discover_vxlan_port_test.go:26]:
```suggestion
var _ = Describe("Modules :: cni-cilium :: hooks :: discover_vxlan_port ::", func() {
```

File: modules/021-cni-cilium/hooks/discover_vxlan_port.go [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+		},
+		Off: { // Virtualization is off
+			TransitionRule{source: 0, target: 4299},
+		},
+	},
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 10},
+	Queue:        "/modules/cni-cilium/set-vxlan-allowed-ports",
```
Comment (apolovov) [inline modules/021-cni-cilium/hooks/discover_vxlan_port.go:91]:
```suggestion
	Queue:        "/modules/cni-cilium/discover_vxlan_port",
```

**Review by apolovov — APPROVED** [review]

**Review by apolovov — APPROVED** [review]

File: docs/documentation/_data/deckhouse-alerts.yml [hunk 9]
```diff
@@ -105,6 +105,19 @@ alerts:
         Cloud data discoverer finds disks in the cloud for which there is no PersistentVolume in the cluster
       severity: "6"
       markupFormat: markdown
+    - name: CniCiliumNonStandardVXLANPortFound
+      sourceFile: modules/021-cni-cilium/monitoring/prometheus-rules/config.yaml
```
Comment (apolovov) [inline docs/documentation/_data/deckhouse-alerts.yml:109]:
```suggestion
      sourceFile: modules/021-cni-cilium/monitoring/prometheus-rules/configmap.yaml
```

**Review by nabokihms — APPROVED** [review]

**Pull request merged**